convert2rhel/checks.py

```diff
@@ -75,81 +93,79 @@ def perform_pre_ponr_checks():
+        return
 
     cmd = [
         "repoquery",
         "--releasever=%s" % system_info.version.major,
+        CDN_URL,
+        "convert2rhel",
     ]
 
-    repo_dir = tempfile.mkdtemp(prefix="convert2rhel_repo.", dir=utils.TMP_DIR)
-    utils.mkdir_p(repo_dir)
-    repo_path = os.path.join(repo_dir, "convert2rhel.repo")
-    store_content_to_file(filename=repo_path, content=repo_content)
-
-    cmd.append("--setopt=reposdir=%s" % repo_dir)
-    cmd.append("convert2rhel")
-
     raw_output_convert2rhel_versions, return_code = run_subprocess(cmd, print_output=False)
 
     if return_code != 0:
         logger.warning(
-            "Couldn't check if this is the latest version of convert2rhel\n"
-            "repoquery failed %s, %s" % (return_code, raw_output_convert2rhel_versions)
+            "Couldn't check if the current installed Convert2RHEL is the latest version.\n"
+            "repoquery failed with the following output:\n%s" % (raw_output_convert2rhel_versions)
         )
         return
 
-    PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
     convert2rhel_versions = re.findall(PKG_NEVR, raw_output_convert2rhel_versions, re.MULTILINE)
-    latest_version = ("0", "0.00", "0")
+    logger.debug("Found %s convert2rhel package(s)" % len(convert2rhel_versions))
+    latest_available_version = ("0", "0.00", "0")
 
+    # This loop will determine the latest available convert2rhel version in the yum repo.
+    # It assigns the epoch, version, and release ex: ("0", "0.26", "1.el7") to the latest_available_version variable.
     for package_version in convert2rhel_versions:
-        # rpm.lableCompare(pkg1, pkg2) compare two kernel version strings and return
-        # -1 if str2 is greater then str1, 0 if they are equal, 1 if str1 is greater the str2
-        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+        # rpm.labelCompare(pkg1, pkg2) compare two package version strings and return
+        # -1 if latest_version is greater than package_version, 0 if they are equal, 1 if package_version is greater than latest_version
+        ver_compare = rpm.labelCompare(package_version[1:], latest_available_version)
         if ver_compare > 0:
-            latest_version = package_version[1:]
+            logger.debug(
+                "...convert2rhel version %s is newer than %s, updating latest known version"
+                % (package_version[1:][1], latest_available_version[1])
+            )
+            latest_available_version = package_version[1:]
+
+    # We hard code 0 for the epoch and release below because latest_version will
+    # contain somthing like "1.el7" for the release but we don't know what the release for
+    # installed_convert2rhel_version will be, if we did not hard code to 0 latest_available_version
+    # will evaluate as later then installed_convert2rhel_version even when the versions
+    # were the same.
```

Comment:
```suggestion
    # After the for loop, the latest_available_version variable will gain the epoch, version, and release 
    # (e.g. ("0" "0.26" "1.el7")) information from the Convert2RHEL yum repo
    # when the versions are the same the latest_available_version's release field will cause it to evaluate as a later version.
    # Therefore we need to hardcode "0" for both the epoch and release below for installed_convert2rhel_version 
    # and latest_available_version respectively, to compare **just** the version field.
```
I liked the previous phrasing better TBH, this one is confusing to me. WDYT @bocekm ?

Reply:
Yeah, the one you suggest is more clear, @danmyway.

Reading it though the functionality should be a bit different. We should compare not only the version but also the release. The reason is that from time to time we need to make a downstream-only release (applying a patch in dist-git only). Doing that results in a bumped up release, for instance `1.1-2.el8`. I'll do a PR review later on to expand more on that.

Reply:
Alright, thank you for the clarification!

Reply:
Yeah thank you both for going over this!